packages/lib/src/project/SimpleProject.js

```diff
@@ -32,12 +33,12 @@ export default class SimpleProject  {
     return new contractClass(proxy.address);
   }
 
-  async upgradeProxy(proxyAddress, contractClass, { initMethod, initArgs, initFrom } = {}) {
+  async upgradeProxy(proxyAddress, contractClass, { contractName, initMethod, initArgs, initFrom, redeployIfChanged } = {}) {
     proxyAddress = toAddress(proxyAddress)
-    const implementation = await this._deployImplementation(contractClass);    
+    const implementation = await this._getOrDeployImplementation(contractClass, contractName, redeployIfChanged);    
     log.info(`Upgrading proxy to new logic contract at ${implementation.address}`)
     const proxy = Proxy.at(proxyAddress, this.txParams)
-    await proxy.upgradeTo(implementation)    
+    await proxy.upgradeTo(implementation) // TODO: Use upgradeToAndCall!
```

Comment:
we should do sth like we are doing in `App#upgradeProxy` right? I mean, if there is no `initMethod` we should call `upgradeTo`, if there is then we should call `upgradeToAndCall`, am I right?